repository/src/main/java/org/alfresco/service/cmr/search/CategoryService.java

```diff
@@ -150,11 +150,19 @@ public enum Depth {IMMEDIATE, ANY};
      */
     @Auditable(parameters = {"storeRef", "aspectName", "pagingRequest", "sortByName", "exactNamesFilter", "alikeNamesFilter"})
     default PagingResults<ChildAssociationRef> getRootCategories(StoreRef storeRef, QName aspectName, PagingRequest pagingRequest, boolean sortByName,
-        Collection<String> exactNamesFilter, Collection<String> alikeNamesFilter)
-    {
+                                                              Collection<String> exactNamesFilter, Collection<String> alikeNamesFilter) {
         return new EmptyPagingResults<>();
     }
 
+    /**
+     * Get a collection of the root categories for an aspect/classification supporting multiple name filters.
+     */
+    @Auditable(parameters = {"storeRef", "aspectName", "exactNamesFilter", "alikeNamesFilter"})
+    default Collection<ChildAssociationRef> getRootCategories(StoreRef storeRef, QName aspectName, Collection<String> exactNamesFilter, Collection<String> alikeNamesFilter) {
+        return null;

```

Comment:
It woluld be better if `emptyList` (or `emptySet` but I see that the impementation returns `List`) was returned here instead of `null`.